src/lib/InstantSearch.js

```diff
@@ -195,6 +195,12 @@ To help you migrate, please refer to the migration guide: https://community.algo
       );
     }
 
+    // The routing manager widget is always added manually at the last position.
+    // By removing it from the last position and adding it back after, we ensure
+    // it keeps this position.
+    // fixes #3148
+    const lastWidget = this.widgets.pop();
```

Comment:
it's a fine fix I guess, but what if the routing manager came to a different key than `this.widgets`? That would solve the issue too no?

Reply:
Yes indeed. I chose to keep it in the `this.widgets` because otherwise it would require to call the lifecycle steps of the routing manager with specific lines of code. I favored the systematic aspect here.

Reply:
now that we are talking about this line anyway; I think it would make sense to call it `_widgets`, since it's exposed on the instantsearch instance, and clearly isn't something that should be touched externally. In a later stage we can use the proper private property when they are supported & polyfilled :)

Reply:
That sounds like a good idea. Do you mind if I do that in another PR because it involves changes in other places too?